Robust.Shared/GameObjects/Systems/SharedTransformSystem.Component.cs

```diff
@@ -512,11 +512,6 @@ public void SetCoordinates(EntityUid uid, TransformComponent xform, EntityCoordi
         if (!xform.Initialized)
             return;
 
-#if DEBUG
-        // If an entity is parented to the map, its grid uid should be null (unless it is itself a grid)
-        if (xform.ParentUid == xform.MapUid)
-            DebugTools.Assert(xform.GridUid == null || xform.GridUid == uid);
-#endif
         var newPosition = xform._parent.IsValid() ? new EntityCoordinates(xform._parent, xform._localPosition) : default;
         var moveEvent = new MoveEvent(uid, oldPosition, newPosition, oldRotation, xform._localRotation, xform, _gameTiming.ApplyingState);
```

Comment:
assuming planet maps still set GridUid, maybe just replce this with
```suggestion
#if DEBUG
        // If an entity is parented to the map, its grid uid should be null (unless it is itself a grid or we have a map-grid)
        if (xform.ParentUid == xform.MapUid)
            DebugTools.Assert(xform.GridUid == null || xform.GridUid == uid || xform.GridUid == xform.MapUid);
#endif
        var moveEvent = new MoveEvent(uid, oldPosition, newPosition, oldRotation, xform._localRotation, xform, _gameTiming.ApplyingState);
```